src/app/shared/components/user-form/user-form.component.spec.ts

```diff
@@ -1,25 +1,79 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { UserFormComponent } from './user-form.component';
+import { FormBuilder } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { FormConfig } from '../../../core/models/formConfig';
 
 describe('UserFormComponent', () => {
   let component: UserFormComponent;
   let fixture: ComponentFixture<UserFormComponent>;
+  let formConfigMock: FormConfig;
+
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       declarations: [ UserFormComponent ],
+      providers: [FormBuilder],
     })
       .compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(UserFormComponent);
     component = fixture.componentInstance;
+    formConfigMock = {
+      title: '',
+      submitted: false,
+      loading: false,
+    };
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should test if the form is valid', () => {
+    expect(component.userForm.valid).toBeFalse();
+
+    component.userForm.controls.userName.setValue('John');
+    component.userForm.controls.userEmail.setValue('asd@gmail.com');
+    component.userForm.controls.password.setValue('1234567');
+    component.userForm.controls.confirmPassword.setValue('1234567');
+
+    expect(component.userForm.valid).toBeTruthy();
+  });
+
+  it('should call onSubmit() method', (() => {
+    const fnc = spyOn(component, 'onSubmit');
+    const form = fixture.debugElement.query(By.css('form'));
+
+    form.triggerEventHandler('ngSubmit', null);
+    expect(fnc).toHaveBeenCalled();
```

Comment:
Check that function in the service was called rather than function in the component